table-store/src/test/java/com/scalar/dl/tablestore/client/partiql/parser/PartiqlParserVisitorTest.java

```diff
@@ -201,4 +204,307 @@ public void parse_InvalidInsertSqlGiven_ShouldThrowIllegalArgumentException() {
+                ImmutableList.of()));
+    assertThat(statements.get(3))
+        .isEqualTo(
+            SelectStatement.create(
+                tableWithAlias,
+                ImmutableList.of(
+                    JacksonUtils.buildCondition("t.col1", "=", TextNode.valueOf("aaa"))),
+                ImmutableList.of()));
+    assertThat(statements.get(4))
+        .isEqualTo(
+            SelectStatement.create(
+                table,
+                ImmutableList.of(JacksonUtils.buildCondition("col1", "=", TextNode.valueOf("aaa"))),
+                ImmutableList.of()));
+    assertThat(statements.get(5))
+        .isEqualTo(
+            SelectStatement.create(
+                table,
+                ImmutableList.of(
+                    JacksonUtils.buildCondition("col1", "=", TextNode.valueOf("aaa")),
+                    JacksonUtils.buildCondition("col2", "!=", BooleanNode.valueOf(false)),
+                    JacksonUtils.buildCondition(
+                        "col3", ">", BigIntegerNode.valueOf(new BigInteger("123"))),
+                    JacksonUtils.buildCondition(
+                        "col4", "<", DecimalNode.valueOf(new BigDecimal("1.23"))),
+                    JacksonUtils.buildCondition(
+                        "col5", ">=", DecimalNode.valueOf(new BigDecimal("1.23e4"))),
+                    JacksonUtils.buildCondition(
+                        "col6", "<=", DecimalNode.valueOf(new BigDecimal("1.23e-4")))),
+                ImmutableList.of()));
+    assertThat(statements.get(6))
+        .isEqualTo(
+            SelectStatement.create(
+                table,
+                ImmutableList.of(
+                    JacksonUtils.buildNullCondition("col1", false),
+                    JacksonUtils.buildNullCondition("col2", true)),
+                ImmutableList.of()));
+    assertThat(statements.get(7))
+        .isEqualTo(
+            SelectStatement.create(
+                table,
+                ImmutableList.of(JacksonUtils.buildCondition("col1", "=", NullNode.getInstance())),
+                ImmutableList.of()));
+    assertThat(statements.get(8))
+        .isEqualTo(
+            SelectStatement.create(
+                table,
+                ImmutableList.of(
+                    JacksonUtils.buildCondition("col1", "=", TextNode.valueOf("aaa")),
+                    JacksonUtils.buildCondition("col2", "=", TextNode.valueOf("bbb")),
+                    JacksonUtils.buildCondition("col3", "=", TextNode.valueOf("ccc"))),
+                ImmutableList.of()));
+  }
+
+  @Test
+  public void parse_InvalidSelectSqlGiven_ShouldThrowIllegalArgumentException() {
+    // Arrange
+    List<String> sqlStatements =
+        ImmutableList.of(
+            "SELECT * FROM tbl AS t AT x",
+            // We should reject below, but we cannot do it for now due to a parser-side issue.
+            // The BY clause is just ignored.
+            // "SELECT * FROM tbl AS t BY x",
```

Comment:
Probably due to a bug in the PartiQL parser, we cannot identify the BY clause in the parser visitor. So, it does not throw an exception, and it is just ignored. I don't think most users notice it since it is not a major functionality, but maybe I should report it.